### PR TITLE
String to number

### PR DESCRIPTION
**Pull request by aehyvari**
As a preparation for combining the arithmetic logics, here are some utilities and refactoring for parsing strings into numbers.

### CODE REVIEW

File: src/common/StringConv.h [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        if (str[0] == '\0') return false;
+
+        bool isInt = true;
+        for (int i = str[0] == '-' ? 1 : 0; str[i] != '\0'; i++) {
+            if (str[i] < '0' or str[i] > '9') {
+                isInt = false;
+                break;
+            }
+        }
+        return isInt;
```
Comment (blishko) [inline src/common/StringConv.h:49]:
```suggestion
        for (int i = str[0] == '-' ? 1 : 0; str[i] != '\0'; i++) {
            if (not isDigit(str[i])) {
                return false;
            }
        }
        return true;
```

**Review by blishko — COMMENTED** [review]
Looks good, left a few comments for simplification.
I trust that you mainly moved the methods from Global.h to respective header, and not make any changes in these methods that were already in the codebase.

File: src/common/StringConv.h [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    }
+
+    bool static inline isRealString(char const *str) {
+        if (str[0] == '\0') return false;
+        enum t_State {
+            S0, S1, S2, S3, S4, S5, S6, S7
+        };
+        auto isDigit = [](char const c) { return c >= '0' and c <= '9'; };
```
[the remark this answers is not in the document]
Comment (aehyvari) [reply src/common/StringConv.h]: Fixed.

**Comment by aehyvari** [comment]
> I trust that you mainly moved the methods from Global.h to respective header, and not make any changes in these methods that were already in the codebase.

Yes, I don't think I changed them, so there shouldn't be anything interesting.

**Review by blishko — APPROVED** [review]

**Pull request merged**